### PR TITLE
Chidera - session Management Function

### PR DESCRIPTION
**Pull request by Zer0SZN**
## What type of PR is this? (check all applicable)

- [x] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Other: Session Management Function && Integration
  
  
## Description of what this Pull Request does!
- **Session Management Function** enables user to initiate a session with peer ID, join session, start session and persist session on page reload in between the time interval of 60mins
## Additional test cases and function for authenticated users!
- Session ID Validation: It validates the user (joining user) session ID by matching the stored session ID initiated by a user (initiator). If the session IDs are not matching toast error “Invalid Session ID” which means even if the session ID is valid and users has reached the max limit of two people it toasts this same error “Invalid Session ID” as expired session ID.
- End session: This function helps user to terminate before or with the time interval(60mins) session for two reasons.
*If the users are done with their session before the timeout interval and wants to terminate the session. It ends the session and remove their stored session data from firebase.
*If the users decides to change language before or within the timeout out interval. Session data stored is removed as well.

## Preview
preview
der/issues/164

## Related Tickets & Documents
- Closes
https://github.com/annonymousauthority/P2PCo

### CODE REVIEW

**Comment by annonymousauthority** [comment]
New Peer session should open the modal that asks for the session name and programming language, this happens first before a generated peer id is generated which is then shown in the final modal.
Please visit the designs, for the flow. You should do the entire flow up until the point the user needs to go to the collab screen

**Review by annonymousauthority — CHANGES_REQUESTED** [review]
I left you a comment

File: development/src/composables/firebaseConfig/config.js
Comment (annonymousauthority) [inline development/src/composables/firebaseConfig/config.js:1]: Please do not touch this file, remove any change you did to this file, thank you

**Review by annonymousauthority — CHANGES_REQUESTED** [review]

**Comment by Zer0SZN** [comment]
> I left you a comment

Aight

**Comment by annonymousauthority** [comment]
Could you integrate this to the main page.js not just show us test components. Thank you

**Comment by Zer0SZN** [comment]
> Could you integrate this to the main page.js not just show us test components. Thank you

**Comment by Zer0SZN** [comment]
> Could you integrate this to the main page.js not just show us test components. Thank you

ok

File: development/src/composables/sessionManagementFunction.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import { generatePeerIdCharacter } from "./peerIdGenerator";
```
Comment (7malikk) [inline development/src/composables/sessionManagementFunction.js:1]: remove comments

File: development/src/composables/sessionManagementFunction.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import { toast } from "react-toastify";
```
Comment (7malikk) [inline development/src/composables/sessionManagementFunction.js:2]: we are no longer using the `react-toastify`
you need to utilize the `errorModal_comp`

Comment (Zer0SZN) [reply development/src/composables/sessionManagementFunction.js:2]: Aight

File: development/src/components/joinSession_comp.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { useState } from "react";
+import { joinPeerSession } from "@/composables/sessionManagementFunction";
+import { toast } from "react-toastify";
```
Comment (7malikk) [inline development/src/components/joinSession_comp.js:3]: utilize `errorModal` for errors

File: development/src/composables/sessionManagementFunction.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import firebase from "firebase/app";
+import "firebase/auth";
+import "firebase/database";
+import { firebaseConfig } from "./firebaseConfig/config";
```
Comment (7malikk) [inline development/src/composables/sessionManagementFunction.js:6]: `firebaseConfig` does not seem to be used in this file? why the import?

File: development/src/composables/firebaseConfig/config.js [hunk 5]
```diff
@@ -24,4 +24,4 @@ const emailProvider = new GoogleAuthProvider();
 // timestamp
 const timestamp = appFirestore.timestamp;
 
-export { appStorage, timestamp, appAuth, appFirestore, emailProvider };
+export { appStorage, timestamp, appAuth, appFirestore, emailProvider, firebaseConfig };
```
Comment (7malikk) [inline development/src/composables/firebaseConfig/config.js:27]: why the export of `firebaseConfig`?

File: development/src/composables/sessionManagementFunction.js (hunk 4 above)
Comment (Zer0SZN) [reply development/src/composables/sessionManagementFunction.js:6]: Will be used to check if users I authenticated and get user session data from firebase DB

File: development/package.json [hunk 6]
```diff
@@ -22,6 +22,7 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-monaco-editor": "^0.52.0",
+    "react-toastify": "^9.1.3",
```
Comment (7malikk) [inline development/package.json:25]: you need to uninstall `react-toastify`

File: development/src/components/PeerOverlay_comp.js [hunk 7]
```diff
@@ -1,83 +1,132 @@
...
+  };
+  if (!isOpen) {
+    return null;
+  }
 
-                <button
-                    className="w-full py-7 px-6 text-white mt-12 rounded-lg bg-[#5F5BD7] text-center font-normal text-lg font-nohemi">
-                    Create Peer Session
-                </button>
+  const theme = useTheme();
```
Comment (7malikk) [inline development/src/components/PeerOverlay_comp.js:49]: this is not the correct implementation of the `useTheme()`

File: development/src/composables/sessionManagementFunction.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
+import { generatePeerIdCharacter } from "./peerIdGenerator";
+import firebase from "firebase/app";
+import "firebase/auth";
+import "firebase/database";
+import { firebaseConfig } from "./firebaseConfig/config";
+import ErrorModal from "@/components/errorModal_comp";
+import { useState } from "react";
+
+firebase.initializeApp(firebaseConfig);
```
Comment (7malikk) [inline development/src/composables/sessionManagementFunction.js:9]: Why is this here?
